k4Gen/src/components/MDIReader.h

```diff
@@ -0,0 +1,70 @@
+#ifndef GENERATION_MDIREADER_H
+#define GENERATION_MDIREADER_H
+
+
+#include "Generation/IHepMCFileReaderTool.h"
+#include "Generation/IVertexSmearingTool.h"
+#include "Generation/IHepMCMergeTool.h"
+
+#include "k4FWCore/DataHandle.h"
+
+#include "GaudiAlg/GaudiAlgorithm.h"
+#include "GaudiKernel/ToolHandle.h"
+
+#include "GaudiKernel/SystemOfUnits.h"
+
+#include "HepMC/IO_GenEvent.h"
```

Comment:
You should use HepMC3 here (see https://github.com/HEP-FCC/k4Gen/blob/main/k4Gen/src/components/HepMCFileReader.h)